core/runtime/op_driver/mod.rs

```diff
@@ -19,9 +19,19 @@ pub use joinset_driver::JoinSetDriver;
 
 pub use self::op_results::OpMappingContext;
 pub use self::op_results::OpResult;
+use self::op_results::PendingOpInfo;
 pub use self::op_results::V8OpMappingContext;
 pub use self::op_results::V8RetValMapper;
 
+#[derive(Default)]
+#[allow(dead_code)] // follow-up PR
+/// Returns a set of stats on inflight ops.
+pub struct OpInflightStats {
+  /// The [`PromiseId`] of any inflight ops by each [`OpId`].
+  ops: Box<[PendingOpInfo]>,
+  supported: bool,
+}
```

Comment:
Could we avoid boxing here if we stored a slice and added a lifetime to `OpInflightStats`?

Reply:
Probably not a big deal though.

Reply:
I figured that these are not collected in a hot loop (and the docstring explicitly warns against that), so we should be OK to allocate.